src/arithmetic/agg_funcs.c

```diff
@@ -197,10 +209,78 @@ AggCtx* Agg_CountFunc() {
 
 //------------------------------------------------------------------------
 
+typedef struct {
+    double percentile;
+    double *values;
+    size_t count;
+} __agg_percCtx;
+
+// This function is agnostic as to percentile method
+int __agg_percStep(AggCtx *ctx, SIValue *argv, int argc) {
+
+    __agg_percCtx *ac = Agg_FuncCtx(ctx);
+
+    // The first argument is the percentile value. This only actually needs to be set once
```

Comment:
According to Neo the last (second) argument is the percentile,
https://neo4j.com/docs/developer-manual/current/cypher/functions/aggregating/#functions-percentiledisc

Reply:
I haven't looked at the code which interprets query statements, but on my local Redis server it seems that argument orders are reversed at some point before populating the `argv` list.

```
127.0.0.1:7777> GRAPH.QUERY imdb "MATCH (a:actor) RETURN perc_d(a.age, .5)"
1) 1) "perc_d(0.500000,a.age)\x00"
   2) "43.000000"
2) 1) "Query internal execution time: 1.316000 milliseconds"
127.0.0.1:7777> GRAPH.QUERY imdb "MATCH (a:actor) RETURN sum(0, 2, 7)"
1) 1) "sum(7.000000,2.000000,0.000000)\x00"
   2) "11862.000000"
2) 1) "Query internal execution time: 0.456000 milliseconds"
```
I can look into where this translation occurs if you would like!